scripts/globals/gobbiemysterybox.lua

```diff
@@ -126,7 +126,7 @@ tpz.mystery.onTrigger = function (player, npc, events)
             player:startEvent(event.DEFAULT, specialDialUsed, adoulinDialUsed, pictlogicaDialUsed, wantedDialUsed, 0, 0, hideOptionFlags, dailyTallyPoints)
         end
     else
-        player:messageSpecial(zones[player:getZoneID()].text.YOU_MUST_WAIT_ANOTHER_N_DAYS, GOBBIE_BOX_MIN_AGE - playerAgeDays)
+        player:messageSpecial(zones[player:getZoneID()].text.YOU_MUST_WAIT_ANOTHER_N_DAYS, GOBBIE_BOX_MIN_AGE - playerAgeDays + 1)
```

Comment:
This will fix the display but it's still not ticking on JST midnight like it should. Instead of changing this I think `playerAgeDays` needs to account for JST midnight since it currently uses `os.time()`. I think you can just switch it to `JstMidnight()`.

Reply:
Based on the data from Nyu, it's not based on JP midnight, but character creation time stamp.

How this was tested:

1. Character created same day and spoke with Gobbie at ~23:14 JST
2. Character spoke with gobbie, "You must wait another 45 days to..."
3. Waited until after JST midnight (12:01 JST)
4. Character spoke with gobbie, "You must wait another 45 days to..."

This debunked my theory that it was based on JST Midnight, seems to confirm that it's based on character creation time stamp, and lends credence to the fix as submitted. (Screenshots below are taken in Time Zone UTC+1)

![image](https://user-images.githubusercontent.com/3996176/98316163-b9ebfd80-1f8e-11eb-8546-dfdbf2d8a13f.png)

![image](https://user-images.githubusercontent.com/3996176/98316099-9923a800-1f8e-11eb-96fa-2488d442c693.png)

Reply:
Oh I misunderstood or misread your first post, my bad! 😫

Reply:
All good, I can see how my last sentence in the PR could be ambiguous. Hopefully this does a better job of painting the picture. :+1: